### PR TITLE
[WIP] Add Link Checker

### PR DESCRIPTION
**Pull request by lindluni**
Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

### CODE REVIEW

File: ci/azure-pipelines.yml [hunk 1]
```diff
@@ -14,6 +14,38 @@ variables:
...
+          displayName: Publish Documentation Artifacts
+
+  - stage: LinkCheck
+    dependsOn: []
+    jobs:
+      - job: LinkCheck
+        pool:
+          vmImage: ubuntu-18.04
+        container:
+          image: btl5037/tox
```
Comment (sykesm) [inline ci/azure-pipelines.yml]: Where's the Dockerfile used to generate this image? I don't see it in the ci folder.

Comment (lindluni) [reply ci/azure-pipelines.yml]: I have it, I just haven't pushed it yet as I am trying to prune the dependencies out of it to make it as small as possible. I am only pushing it here to test. Should have opened as a draft.

Comment (lindluni) [reply ci/azure-pipelines.yml]: It's also going to artifactory at the end, I'm not going to host it

Comment (denali49) [reply ci/azure-pipelines.yml]: FYI I ran a free link checker as a sanity check on the Fabric docs and showed ~40 out of 2k+ links broken.  I ran a cursory check of a few and it seemed pretty accurate.  I'll bring this up with the docs group and we'll start fixing them under a separate PR(s).  Thanks for doing this Brett.

Comment (denali49) [reply ci/azure-pipelines.yml]: BTW, deadlinkchecker.com is the sanity check tool I ran.

Comment (lindluni) [reply ci/azure-pipelines.yml]: @denali49 Yea, I think there are only a handful that aren't real (mostly example URL's) which I need to figure out how to explicitly ignore a link in sphinx

File: docs/source/conf.py [hunk 2]
```diff
@@ -218,3 +219,11 @@ def setup(app):
...
 # Increase the linkcheck timeout to 5 seconds
 linkcheck_timeout = 5
+
+def entire_domain(host):
+    return r"http.?://" + re.escape(host) + r"($|/.*)"
+
+linkcheck_ignore = [
+    # Skip github.com until proper rate-limiting or authenticated requests are available: https://github.com/sphinx-doc/sphinx/issues/7388
+    entire_domain("github.com")
+]
```
Comment (lindluni) [inline docs/source/conf.py]: Newline

File: docs/source/developapps/application.md [hunk 3]
```diff
@@ -115,7 +115,7 @@ transaction proposal to the right peer nodes in the network using the
 options](./connectionoptions.html).
 
 Spend a few moments examining the connection
-[profile](https://github.com/hyperledger/fabric-samples/blob/{BRANCH}/commercial-paper/organization/magnetocorp/gateway/networkConnection.yaml)
+[profile](#https://github.com/hyperledger/fabric-samples/blob/{BRANCH}/commercial-paper/organization/magnetocorp/gateway/networkConnection.yaml)
```
Comment (lindluni) [inline docs/source/developapps/application.md]: Remove the `#` no longer needed with the `ignore domain` directive

File: .readthedocs.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+build:
```
Comment (lindluni) [inline .readthedocs.yml]: This file can be removed once all of this is fixed and we can set the default language version in READTHEDOCS admin panel

**Comment by lindluni** [comment]
Need to address the formatting issues preventing the docs from properly compiling now that we've had to move to newer versions of Sphinx

**Comment by joealewine** [comment]
@btl5037 Is there a certain kind of formatting error we can address in bulk? Because the huge number of issues we were talking about formatting wise doesn't seem possible to address given our bandwidth.

**Comment by lindluni** [comment]
@joealewine I am going to try and investigate it tonight and see if I can fix one of the docs and hopefully pinpoint the compilation issue.

**Pull request closed without merging**